### ISSUE TITLE
Cannot programmatically show keyboard after hiding it

### ISSUE THREAD

**Issue by AungThiha** [verygoodsecurity/vgs-collect-android#200]
**Describe the bug**
After hiding the keyboard programmatically, it cannot be shown programmatically again. Tapping on the vgsEditText shows the keyboard. Just can't do that programmatically

**Additional context**
When we call `vgsForm.asyncSubmit("/post", HTTPMethod.POST)`, we wanna disable the vgsEditText to prevent the user from typing or at least hide the keyboard and show some overlay. Then, when we get back error from vgs inbound, we wanna show the keyboard automatically.

**To Reproduce**

AndroidManifest.xml
```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.ta.vgs">

    <application
        android:allowBackup="true"
        android:icon="@mipmap/ic_launcher"
        android:label="@string/app_name"
        android:roundIcon="@mipmap/ic_launcher_round"
        android:theme="@style/Theme.MaterialComponents.DayNight.NoActionBar">
        <activity android:name=".SimpleKeyboardTestActivity"
            android:windowSoftInputMode="adjustResize|stateVisible"
            />
    </application>

</manifest>
```

activity_simple_keyboard_test.xml
```xml
<?xml version="1.0" encoding="utf-8"?>
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:background="#6C3F72"
    android:orientation="vertical"
    android:padding="16dp">

    <com.verygoodsecurity.vgscollect.widget.VGSEditText
        android:id="@+id/vgsEditText"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        app:fieldName="pin" />

    <Button
        android:id="@+id/button"
        android:text="hide/show"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"/>

</LinearLayout>
```

SimpleKeyboardTestActivity.kt
```kotlin
class SimpleKeyboardTestActivity : AppCompatActivity() {

    private lateinit var vgsEditText: VGSEditText
    private lateinit var button: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_simple_keyboard_test)

        vgsEditText = findViewById(R.id.vgsEditText)
        button = findViewById(R.id.button)

        button.setOnClickListener {
          
            vgsEditText.isEnabled = false
            
            // this is just to simulate vgsForm.asyncSubmit("/post", HTTPMethod.POST) and its error response
            Thread {
                Thread.sleep(1000)
                runOnUiThread {
                    vgsEditText.isEnabled = true
                    showKeyboard()
                }
            }.start()
        }

        showKeyboard()
    }

    private fun showKeyboard() {
        vgsEditText.apply {
            clearFocus()
            requestFocus()
            performClick()
            val inputManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            inputManager.showSoftInput(this, InputMethodManager.SHOW_FORCED)
        }
        window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE)
    }

}
```

**Attempt 1:** I have played around using only one of or combination of two of or three of  the lines in this func
```kotlin
private fun showKeyboard() {
    vgsEditText.apply {
        clearFocus()
        requestFocus()
        performClick()
        val inputManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        inputManager.showSoftInput(this, InputMethodManager.SHOW_FORCED)
    }
    window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE)
}
```

**Attempt 2:** I have tried hiding the keyboard instead of disabling the vgsEditText
```kotlin
val inputManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
inputManager.hideSoftInputFromWindow(vgsEditText.windowToken, 0)
```

**Attempt 3:** I have tried wrapping content of `showKeyboard()` inside `post`
Please note that if I put `setSoftInputMode` inside `post`, even showing keyboard initially doesn't work. That's why it's not in `post` in this example.
```kotlin
private fun showKeyboard() {
    vgsEditText.apply {
        post {
            clearFocus()
            requestFocus()
            performClick()
            val inputManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            inputManager.showSoftInput(this, InputMethodManager.SHOW_FORCED)
        }
        window?.setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE)
    }
}
```

**Expected behavior**
The keyboard should be shown again when the vgsEditText is re-enabled and `showKeyboard` func is called 

**Note**
If I replace `com.verygoodsecurity.vgscollect.widget.VGSEditText` with normal `EditText`, it works

**Environment (please complete the following information):**
 - SDK Version: 1.6.9
 - Android Version: 11
 - Device: Samsung Galaxy S20

**Comment by AungThiha** [comment]
**Attempt 4**
I was wondering if this was due to hopping back to main thread from a background thread. So, I tried adding a `enable` button and showing keyboard from it. Still no luck
```kotlin
buttonDisable.setOnClickListener {
    vgsEditText.isEnabled = false
}

buttonEnable.setOnClickListener {
    vgsEditText.isEnabled = true
    showKeyboard()
}
```

**Attempt 5**
I have also tried hiding the keyboard instead of disabling the vgsEditText with this approach too. Doesn't work either.
```kotlin
buttonDisable.setOnClickListener {
    val inputManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
    inputManager.hideSoftInputFromWindow(vgsEditText.windowToken, 0)
}

buttonEnable.setOnClickListener {
    showKeyboard()
}

```

**Comment by Endorf** [comment]
Thank you for mentioning us about this issues. We will take a look on it.

**Comment by DmytroDm** [comment]
@AungThiha Can you try this version of `showKeyboard()/hideKeyboard()` functions:
 ```
private fun showKeyboard() {
    vgsEditText.isEnabled = true
    vgsEditText.requestFocus()
    val im = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
    im.toggleSoftInput(InputMethodManager.SHOW_FORCED,0)
 }

private fun hideKeyboard() {
    vgsEditText.isEnabled = false
    vgsEditText.clearFocus()
    val im = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
    im.hideSoftInputFromWindow(vgsEditText.windowToken,0)
}
```
As you can see I've used `toggleSoftInput` and not `showSoftInput` in `showKeyboard()`, unfortunately, you can't use `showSoftInput` because it requires reference to focused view, but as we use wrappers for our custom views you can't get this reference.

Also, we will add `showKeyboard()/hideKeyboard()` functions to our custom views in the next releases, so it will give the ability to control the keyboard from our views.

**Comment by AungThiha** [comment]
@DmytroDm thanks, it works.

One minor case though. If the keyboard is already shown and if I use `showKeyboard`, the keyboard gets hidden and there's no easy way to check if a keyboard is visible.

**Comment by DmytroDm** [comment]
@AungThiha Hi, we had released a new version 1.6.10. We have added `showKeyboard()hideKeyboard()` functions to all VGS edit texts, so now you can use it like this:
```
vgsEditText.requestFocus()
vgsEditText.showKeyboard()
```
This will also fix this case that you described.

**Comment by AungThiha** [comment]
Works like a charm. Thanks, @DmytroDm
